lib/insights_cloud/async/insights_full_sync.rb

```diff
@@ -2,17 +2,19 @@
 
 module InsightsCloud
   module Async
-    class InsightsFullSync < ::ApplicationJob
+    class InsightsFullSync < ::Actions::EntryAction
       include ::ForemanRhCloud::CloudAuth
 
-      def perform
+      def plan
```

Comment:
You should also change the way the task is being triggered in https://github.com/theforeman/foreman_rh_cloud/blob/master/app/controllers/insights_cloud/tasks_controller.rb#L4

otherwise there's the following error:
```
NoMethodError (undefined method `perform_later' for InsightsCloud::Async::InsightsFullSync:Class):
```